lib/blockchain.go

```diff
@@ -1674,39 +1622,137 @@ func (bc *Blockchain) Snapshot() *Snapshot {
 // invalidate and chop off the headers corresponding to those blocks and
 // their ancestors so the two generally stay in sync.
 func (bc *Blockchain) blockTip() *BlockNode {
-	var tip *BlockNode
-
-	if len(bc.bestChain) == 0 {
-		return nil
-	}
-
-	tip = bc.bestChain[len(bc.bestChain)-1]
-
-	return tip
+	return bc.blockIndex.GetTip()
 }
 
 func (bc *Blockchain) BlockTip() *BlockNode {
 	return bc.blockTip()
 }
 
+// TODO: this won't work for now. Need to figure out how to handle this.
 func (bc *Blockchain) BestChain() []*BlockNode {
-	return bc.bestChain
+	panic("BestChain not supported.")
```

Comment:
Another thing you could do is add an argument like "minHeight" or something to this, and then the things that need it can get however many blocks in the past they need. This is what the hot feed needs and I see you kinda do it in the db_utils version of GetBestChain.

Reply:
I actually don't see this function used in the hot feed routine. We need to replace the usages of this in the rosetta repo though.